gnovm/cmd/gno/lint.go

```diff
@@ -172,27 +172,36 @@ func catchRuntimeError(pkgPath string, stderr io.WriteCloser, action func()) (ha
 			return
 		}
 		hasError = true
-		switch verr := r.(type) {
-		case *gno.PreprocessError:
-			err := verr.Unwrap()
-			fmt.Fprint(stderr, issueFromError(pkgPath, err).String()+"\n")
-		case scanner.ErrorList:
-			for _, err := range verr {
-				fmt.Fprint(stderr, issueFromError(pkgPath, err).String()+"\n")
-			}
-		case error:
-			fmt.Fprint(stderr, issueFromError(pkgPath, verr).String()+"\n")
-		case string:
-			fmt.Fprint(stderr, issueFromError(pkgPath, errors.New(verr)).String()+"\n")
-		default:
-			panic(r)
-		}
+		printRuntimeError(r, pkgPath, stderr)
 	}()
 
 	action()
 	return
 }
 
+func printRuntimeError(r interface{}, pkgPath string, stderr io.WriteCloser) {
```

Comment:
Is this new function creation necessary? It seemed better to handle the runtime error within the existing `catchRuntimeError` function.

Reply:
The reason I created the function is because I calling it recursively on this case:
When analysing several files we will panic with an array of errors 
[]error or after the change you proposed a multierr.

knowing that inside each of these error there is a scanner.Errorlist that needs to be handled we have 2 options
- with []error (multierr) first time we will enter on the case and call recursively `printRuntimeError` function again for each element (scanner.ErrorList)
```go
case []error:
	for _, err := range verr {
		// recursive call to handle specifically each error type ex: scanner.ErrorList
		printRuntimeError(err, pkgPath, stderr)
	}
```
Second time we will enter
```go
case scanner.ErrorList:
for _, err := range verr {
	fmt.Fprint(stderr, issueFromError(pkgPath, err).String()+"\n")
}
```

- Second possibility is to keep the function as it were but having an extra for loop something like
```go
case []error (multiErr):
	for _, errL := range verr {
		errorList := errL.(scanner.ErrorList)
		for _, err := range errorList {
                         fmt.Fprint(stderr, issueFromError(pkgPath, errors.New(err)).String()+"\n")
                 }
	}
```

Reply:
implemented second possibility on 18fe895c2e5c05af7a5f3ff804ac9da746bd39df
Don't know if you consider it better